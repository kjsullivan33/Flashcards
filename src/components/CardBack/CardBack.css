.card-front {
  width: 500px;
  border-radius: 20px;
  margin-bottom: 30px;
}

.input-question {
  margin-top: 80px;
}