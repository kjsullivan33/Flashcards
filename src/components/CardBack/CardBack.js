import React from 'react';
import './CardBack.css';


const cardBack = (props) => {
  return (
    <div className="card-back">
      <h2>Answer</h2>
      <div>{props.answer}</div>
    </div>
  );
}



export default cardBack;