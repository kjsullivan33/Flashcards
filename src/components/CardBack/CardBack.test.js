import React from 'react';
import CardBack from './CardBack';
import { shallow } from 'enzyme';
import '../../setupTest';
import '../../tempPolyfills';

describe('CardBack', () => {

  const cardBack = shallow(<CardBack />);

  it('renders properly', () => {
    expect(cardBack).toMatchSnapshot();
  });

  //initializes state
  it('initializes an answer in `state`', () => {
    expect(cardBack.state()).toEqual({ answer: '' });
  });
  //sets state

  describe('when typing into the answer input', () => {
    const answer = 'Kristen Sullivan';

    beforeEach(() => {
      cardBack.find('.input-answer').simulate('change', { target: { value: answer } });
    });

    it('updates the question in `state`', () => {
      expect(cardBack.state().answer).toEqual(answer);
    });
  });
});