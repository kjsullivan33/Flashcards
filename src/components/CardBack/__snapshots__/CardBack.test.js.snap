// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardBack renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardBack />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <form>
          <label>
                    Answer: 
          </label>
          <input
                    className="input-answer"
                    onChange={[Function]}
                    type="text"
                    value=""
          />
</form>,
        <div />,
      ],
      "className": "card-back",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label>
              Answer: 
</label>,
            <input
              className="input-answer"
              onChange={[Function]}
              type="text"
              value=""
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Answer: ",
            },
            "ref": null,
            "rendered": "Answer: ",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "input-answer",
              "onChange": [Function],
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "form",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
        },
        "ref": null,
        "rendered": "",
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <form>
            <label>
                        Answer: 
            </label>
            <input
                        className="input-answer"
                        onChange={[Function]}
                        type="text"
                        value=""
            />
</form>,
          <div />,
        ],
        "className": "card-back",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label>
                Answer: 
</label>,
              <input
                className="input-answer"
                onChange={[Function]}
                type="text"
                value=""
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Answer: ",
              },
              "ref": null,
              "rendered": "Answer: ",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "input-answer",
                "onChange": [Function],
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "form",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
          },
          "ref": null,
          "rendered": "",
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
